### ISSUE TITLE
References as Node type?

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/RiverscapesXML#218]
# The Request / Challenge
In many riverscape projects, it would be nice to provide references. Typical references to want to include: 
- References as related reports to cite (e.g. background papers, reports, datasources etc.)
- The formal data citation (e.g. with DOI) for the paper, report, publication associated with the RS project
- How to cite *this* Riverscape Project (e.g. warehouse URL - https://northarrowresearchlabs.github.io/riverscapes/#/Anabranch/fa668588-10ff-4d18-9a67-015418458073) or if we ever [mint DOIs](https://riverscapes.xyz/Data_Warehouses/#advantages) for RS Projects).

## Functionality
I see the same feature showing up in at least three places within the RS project:
1. Right Click on Project Node
![image](https://user-images.githubusercontent.com/9723905/82121273-eabe4900-9748-11ea-94b1-42b2f0b9cfa6.png)

2. Right Click on Dataset Node
3. Within warehouse
![image](https://user-images.githubusercontent.com/9723905/82121330-7041f900-9749-11ea-988b-c46c2fe450d3.png)

What actually shows up in a dialog when accessed from the above three places should be the sorted and formatted list of references (remember [what we did with Zotero here](https://riverscapes.xyz/Science/publications.html) @philipbaileynar ). There should also be a download button. Optionally a PDF might be included.

## Question/Suggestions
I leave it to @MattReimer and @philipbaileynar to decide best way to achieve. I'm looking for your guideance on best way to include these in
- `project_rs.xml` - *do they need their own special tag*?
- business logic -*

In the project directory, it makes sense for these records  to sit in one place (in their own `references` folder maybe), with one *.ris file and whatever PDFs. 

### RIS Format
The [RIS file format](https://en.wikipedia.org/wiki/RIS_(file_format) seems to be the most common format for storing this and is recognized by almost every citation manager software and website (e.g. Zotero, Endnote, BibTex, tec.). It is just a simple text format with tags, that allows multiple records to be stored in one text file (denoted by the `TY` tag for type of record). You can easily export and import RIS files from Zotero, and easily append entries. There are also tons of examples for how to read RIS to a formatted output (e.g. HTML, MD, RTF etc.)

**Comment by joewheaton** [comment]
I still want to do this @MattReimer and @philipbaileynar, but it seems like this is something to deal with in RivrscapesXML so I moved it from RAVE Addin

**Comment by MattReimer** [comment]
Yup. Definitely an XML thing. We might be able to extend some well known reference XML format. We can definitely discuss this further.
